posthog/queries/person_query.py

```diff
@@ -95,6 +97,9 @@ def get_query(self, prepend: Optional[Union[str, int]] = None, paginate: bool =
         search_clause, search_params = self._get_search_clause(prepend=prepend)
         distinct_id_clause, distinct_id_params = self._get_distinct_id_clause()
         email_clause, email_params = self._get_email_clause()
+        filter_future_persons_query = (
+            "and max(created_at) < now() + interval '1 minute'" if filter_future_persons else ""
```

Comment:
1. We want to check this specific row not previous versions of this person, which might have a created_at in the future. This can happen for example, when we merge (we'd use the older created_at going forward).
2. Generally we use sth like no longer than 1 day in the future because of potential mess-ups with timezones / clocks (could be on their side mess-up if they provide timestamps), would make things easier to debug, it also feels fine to show persons up to 1 day in the future.
```suggestion
            "and argMax(created_at, version) < now() + interval '1 day'" if filter_future_persons else ""
```